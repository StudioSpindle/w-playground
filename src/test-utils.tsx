import React from 'react';
import { render, RenderOptions, RenderResult, Queries } from '@testing-library/react';
import { MuiThemeProvider } from '@material-ui/core/styles';
import { MemoryRouter } from 'react-router-dom';
import themeSpec from './themes/default';
import { PlaygroundProvider } from './playground-context';
import '@testing-library/jest-dom/extend-expect';

const AllTheProviders: React.FunctionComponent<{}> = ({ children }) => (
  <PlaygroundProvider>
    <MuiThemeProvider theme={themeSpec.theme}>
      <MemoryRouter>{children}</MemoryRouter>
    </MuiThemeProvider>
  </PlaygroundProvider>
);

const customRender = (ui: React.ReactElement, options?: RenderOptions<Queries>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
