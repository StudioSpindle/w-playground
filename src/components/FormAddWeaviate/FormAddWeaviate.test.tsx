import * as React from 'react';
import { render } from '../../test-utils';
import FormAddWeaviateUrl from './FormAddWeaviate';

it('matches snapshot', () => {
  const container = render(
    <FormAddWeaviateUrl handleSubmit={jest.fn()} handleUrlChange={jest.fn()} />,
  );
  expect(container).toMatchSnapshot();
});
