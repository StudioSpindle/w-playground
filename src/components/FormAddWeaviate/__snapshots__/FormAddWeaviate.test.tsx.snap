// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        id="formAddWeaviate"
      >
        <div
          class="MuiBox-root MuiBox-root-3"
        >
          <div
            class="MuiBox-root MuiBox-root-4"
          >
            <div
              class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
            >
              <div
                class="MuiFormControl-root MuiTextField-root makeStyles-input-1"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
                  data-shrink="false"
                  for="weaviateUri"
                  id="weaviateUri-label"
                >
                  Weaviate URL
                  <span
                    class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                >
                  <input
                    aria-describedby="weaviateUri-helper-text"
                    aria-invalid="false"
                    autocomplete="weaviateUri"
                    class="MuiInputBase-input MuiInput-input"
                    id="weaviateUri"
                    name="weaviateUri"
                    required=""
                    type="text"
                    value=""
                  />
                </div>
                <p
                  class="MuiFormHelperText-root Mui-required"
                  id="weaviateUri-helper-text"
                >
                  i.e. http://localhost:8080
                </p>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-72"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                Connect Weaviate
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      id="formAddWeaviate"
    >
      <div
        class="MuiBox-root MuiBox-root-3"
      >
        <div
          class="MuiBox-root MuiBox-root-4"
        >
          <div
            class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          >
            <div
              class="MuiFormControl-root MuiTextField-root makeStyles-input-1"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
                data-shrink="false"
                for="weaviateUri"
                id="weaviateUri-label"
              >
                Weaviate URL
                <span
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-describedby="weaviateUri-helper-text"
                  aria-invalid="false"
                  autocomplete="weaviateUri"
                  class="MuiInputBase-input MuiInput-input"
                  id="weaviateUri"
                  name="weaviateUri"
                  required=""
                  type="text"
                  value=""
                />
              </div>
              <p
                class="MuiFormHelperText-root Mui-required"
                id="weaviateUri-helper-text"
              >
                i.e. http://localhost:8080
              </p>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-72"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Connect Weaviate
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
